content/WebAuthn/WebAuthn_Developer_Guide/user-presence-vs-user-verification.adoc

```diff
@@ -0,0 +1,29 @@
+
+User **verification** serves to ensure through the presentation of a shared secret that the person authenticating to a service is in fact who they say they are for the purposes of that service. Or, in the words of the W3C recommendation, "...a user and the user’s client (containing at least one authenticator) work in concert to cryptographically prove to [that service, the RP] that the user controls the private key credential associated with a previously-registered public key credential". The user manipulates the authenticator that manages the key material in such a way as to invoke the ``authenticatorMakeCredential`` and ``authenticatorGetAssertion`` operations and the authenticator authorizes those operations locally. A shared secret can take various forms, such as password, PIN, fingerprint, public key credential, etc. The point is to distinguish one user from any other, i.e., uniquely identify the user.
+
+With user **presence**, the intent is not to identify the user, but to ensure that a user is in control of the key. Presumably that user is the one previously identified, but unless a PIN was required when the credential was created and again at authorization, it could be any individual who is physically present at the location where these ceremonies are performed.
+
+Note that the term "authorization gesture" is ambiguous: it can refer either to user presence or user verification.
+
+Registration requires any of the following:
+
+* User presence - this most basic configuration is nothing more than a FIDO2 call that prompts the user to touch the security key or interact in some way with the authenticator.
+* User verification - a FIDO2 call that invokes the ``authenticatorMakeCredential`` and ``authenticatorGetAssertion`` operations, which the authenticator authorizes locally.
+* User presence + user verification.
+
+The RP has the following options for ``userVerification`` when creating credentials:
+
+*DISCOURAGED*: This value indicates that the RP does not want user verification employed during the operation (for example, to minimize disruption to the user interaction flow).
+
+*PREFERRED*:	This value indicates that the RP prefers user verification for the operation if possible, but will not fail the operation if the response does not have the ``AuthenticatorDataFlags.UV`` flag set.
+
+*REQUIRED*: Indicates that the RP requires user verification for the operation and will fail the operation if the response does not have the ``AuthenticatorDataFlags.UV flag`` set.
+
+
+== Use Cases ==
+* User presence is appropriate for second factor authentication (2FA)
+* User verification is not recommended for 2FA because the user will have already entered a shared secret with the server (password) sent over the network
```

Comment:
I think it would read better like this; you're welcome to ignore this if you disagree.

```suggestion
* User verification is not recommended for 2FA because the user will have already entered a shared secret (password) sent to the server over the network
```

Reply:
Oh good. This does indeed clarify; I didn't write the text in the first place, and I did wonder what was meant.